packages/models.py

```diff
@@ -0,0 +1,155 @@
+import enum
+import uuid as uuid
+from django.db import models
+
+
+class City(models.Model):
+    CENTER = 'Central'
+    NORTH = 'Northern'
+    SOUTH = 'Southern'
+
+    AREA_CHOICES = [
+        (CENTER, 'Central'),
+        (NORTH, 'Northern'),
+        (SOUTH, 'Southern')
+    ]
+
+    name = models.CharField(max_length=32, unique=True, primary_key=True)
+    area = models.CharField(max_length=16, choices=AREA_CHOICES)
+
+    def __str__(self):
+        return self.name
+
+
+class Address(models.Model):
+    street_name = models.CharField(max_length=32)
+    street_number = models.PositiveSmallIntegerField()
+    city = models.ForeignKey(City, on_delete=models.CASCADE)
+
+    def __str__(self):
+        return f"{self.city}, {self.street_name} {self.street_number}"
+
+
+class Partner(models.Model):
+    name = models.CharField(max_length=64, unique=True, primary_key=True)
+    rates = models.CharField(max_length=32, default='0')
+    address = models.ForeignKey(Address, on_delete=models.CASCADE)

```

Comment:
This might be a problem. So what we might see is that when the `Address` model is deleted, the partner will also be deleted. See about [`on_delete`](https://docs.djangoproject.com/en/3.1/ref/models/fields/#django.db.models.ForeignKey.on_delete)

Reply:
Yeah, makes sense, `PROTECT` seems to be appropriate

Reply:
Notice you have a lot of `CASCADE` in other locations which seem a bit risky, for example `Package.origin`. Go over them as well and fix where you feel its necessary.

Reply:
Protect 
```python
    origin = models.ForeignKey(Address, on_delete=models.CASCADE, related_name='origin')
    destination = models.ForeignKey(Address, on_delete=models.CASCADE, related_name='destination')
```

Reply:
Yeah, changed them also

Reply:
Done